pinot-spi/src/main/java/org/apache/pinot/spi/config/table/IndexingConfig.java

```diff
@@ -53,6 +53,8 @@
   private SegmentPartitionConfig _segmentPartitionConfig;
   private boolean _aggregateMetrics;
   private boolean _nullHandlingEnabled;
+  private boolean _optimizeDictionaryEnabled;
```

Comment:
Let's add some comments here explaining the configs because these 2 configs are not very self-explained